applications/StructuralMechanicsApplication/custom_elements/prestress_membrane_element.hpp

```diff
@@ -115,169 +115,122 @@ namespace Kratos
     Vector mDetJ0;
     double mTotalDomainInitialSize;
 
-    std::vector< array_1d<double, 3> > mStrainsVector;	      //container of Strain // TODO is this needed?
-    std::vector< array_1d<double, 6> > mStressesVector;	      //container of Stress // TODO is this needed?
-    std::vector< array_1d<double, 6> > mCauchyStressesVector;	//container of Stress // TODO is this needed?
-
     std::vector< Matrix >              mGVector;
     std::vector< array_1d<double, 3> > mGab0;
-    std::vector< array_1d<double, 3> > mG1;                   // Base vector 1 in updated reference configuration
-    std::vector< array_1d<double, 3> > mG2;                   // Base vector 2 in updated reference configuration
-
-    // Using this variable is a potential bug if the element is not used in formfinding!
-    // In the future this should be a Processinfo Variable (e.g. FROMFINDING_STEP), which
-    // is set by the Fromfinding Strategy
-    // The element can then check if this Var is set and use it accordingly
-    unsigned int mStep;                                       // Simulation step for formfinding
 
     bool mAnisotropicPrestress;                               // determines if isotropic or anisotropic prestress is applied
     std::vector< array_1d<double, 3> > mG1Initial;            // Base vector 1 in initial reference configuration
     std::vector< array_1d<double, 3> > mG2Initial;            // Base vector 2 in initial reference configuration
-    std::vector< array_1d<double, 3> > mG3Initial;            // Base vector 2 in initial reference configuration
+    std::vector< array_1d<double, 3> > mG3Initial;            // Base vector 3 in initial reference configuration
 
 
     void CalculateAll(
       MatrixType& rLeftHandSideMatrix,
       VectorType& rRightHandSideVector,
       const ProcessInfo& rCurrentProcessInfo,
-      bool CalculateStiffnessMatrixFlag,
-      bool CalculateResidualVectorFlag);
+      const bool& rCalculateStiffnessMatrixFlag,
+      const bool& rCalculateResidualVectorFlag);
 
     void CalculateAndAddKm(
-      Matrix& K,
-      Matrix& msB,
-      Matrix& msD,
-      double weight);
+      Matrix& rK,
```

Comment:
One question, due to the fact that the dimension is known, can BoundedMatrix be used?